index.js

```diff
@@ -31,23 +32,39 @@ exports.handler = async event => {
     // kinesis downloads/impressions to be logged in batch
     const kinesisRecords = []
 
+    // only assumerole once for dynamodb access
+    const ddbClient = await dynamo.client()
+
+    // concurrently lookup arrangement json
```

Comment:
Moved loading arrangements (from S3 or DDB) upfront.  Just in case DDB throttles a read request, we can throw that error and retry the entire batch, long before starting to process the bytes-downloaded records (the `await ByteRange.load(...)`)